Casks/stack.rb

```diff
@@ -0,0 +1,25 @@
+cask "stack" do
+  version "2.11.1-20220906"
+  sha256 "d5b92e5f0070fbd1f92c36be58e0c979efcceb79b685dd7e665ebf165d70a82c"
+
+  url "https://filehosting-client.transip.nl/packages/stack/v#{version}/macos/stack-v#{version}.dmg"
+  name "STACK"
+  desc "Personal online hard drive to store, view and share files"
+  homepage "https://www.transip.nl/stack/"
+
+  livecheck do
+    url "https://filehosting-client.transip.nl/packages/stack/"
+    regex(%r{href=["'].*?stack/v?((?:2)(?:\.\d+){2}(?:[-_]\d+))["' >]}i)
```

Comment:
```suggestion
    regex(%r{href=["'](?:.*?/)?v?(2(?:[.-]\d+)+)["' >]}i)
```

This can be simplified a bit and brought more in line with prevailing regex standards.

* This is a directory listing page dedicated to stack, so we don't need to specifically match the leading `stack/` in the `href` attribute URLs
* Using `(?:2)` in the capture group is no different than `2`
* Restricting versions to three parts (e.g., 1.2.3) makes this more brittle (e.g., it would miss a version with fewer or more version parts), so we should avoid it unless there's a specific reason
* In this case, we can simply use `(?:[.-]\d+)+` to match a format like `1.2.3-20230406` (it will also match `1.2.3` without the date suffix, if a version like that ever appears)

Reply:
Agree, this makes more sense and looks better. Thanks for clarifying.